packages/code-du-travail-frontend/src/search/AddressQuery.js

```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import PropTypes from "prop-types";
+import {
+  searchAnnuaireByQuery,
+  searchAnnuaireByCoord
+} from "../annuaire/adresse.service";
+import styled from "styled-components";
+import AsyncFetch from "../lib/AsyncFetch";
+import { AddressResults } from "./AddressResults";
+
+export class AddressQuery extends React.Component {
+  static propTypes = {
+    query: PropTypes.string.isRequired,
+    coord: PropTypes.object
+  };
+  static defaultProps = {
+    coord: null
+  };
+
+  shouldComponentUpdate(nextProps) {
+    if (
+      nextProps.query === this.props.query &&
+      nextProps.coord === this.props.coord
+    ) {
+      return false;
+    }
+    return true;
+  }
+
+  fetch = () => {
+    const { query, coord } = this.props;
+    if (coord) {
+      return searchAnnuaireByCoord(coord);
+    }
+    return searchAnnuaireByQuery(query);
```

Comment:
Might be simplified to 

```js
return searchAnnuaireByCoord(coord) || searchAnnuaireByQuery(query);
```

`searchAnnuaireByCoord` and `searchAnnuaireByQuery` return false if no argument is given.

Reply:
the pattern is nice, but searchAnnuaireByCoord and searchAnnuaireByQuery should return Promise. returning a boolean will make an error since fetch caller want a thenable object